app/component/map/VehicleMarkerContainer.js

```diff
@@ -127,9 +128,17 @@ const connectedComponent = connectToStores(
       );
       vehiclesFiltered = Object.fromEntries(filtered);
     }
+    // if you keep the UI open for a long time then trips that have finsished accumulated on the screen
+    // this removes anything that hasn't had an update in 3 minutes
+    const vehiclesWithRecentUpdates = Object.entries(vehiclesFiltered).filter(
+      ([, message]) => {
+        const threeMinutesAgo = moment().subtract(180, 'seconds');
+        return moment.unix(message.timestamp).isAfter(threeMinutesAgo);
```

Comment:
At least with the vehicle position feeds we use, the message.timestamp value comes from the feed instead of being determined in client. If either client or the feed clock is not in sync, vehicles might be filtered even though they've actually received recent updates.

Reply:
I'm not sure I know what the feed clock is. Are you saying that the timestamp doesn't refer to the time when the message was sent but some other one?

If the client clock is off, wouldn't that also cause a lot of other issues?

Reply:
One solution to this: adding a `receivedAt` timestamp for each message and then filter by that.

Reply:
Currently, the timestamp refers to the time the message was sent but I think receivedAt timestamp would fix this potential issue.

Reply:
In case I wasn't totally clear: I meant adding this in the client.

Reply:
The receivedAt timestamp should be generated at the client, if you mean that. Timestamp is generated at the origin of the data.